AmplifyPlugins/API/AWSAPICategoryPlugin/Core/AppSyncListDecoder.swift

```diff
@@ -0,0 +1,51 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Amplify
+
+public struct AppSyncListDecoder: ModelListDecoder {
+    public static func shouldDecode(decoder: Decoder) -> Bool {
+        guard let json = try? JSONValue(from: decoder) else {
+            return false
+        }
+        return shouldDecode(json: json)
+    }
+
+    static func shouldDecode(json: JSONValue) -> Bool {
+        if case let .object(jsonObject) = json,
+           case .array = jsonObject["items"] {
+            return true
+        }
+
+        return false
+    }
+
+    public static func getListProvider<ModelType: Model>(modelType: ModelType.Type,
+                                                         decoder: Decoder) throws -> AnyModelListProvider<ModelType> {
+        let json = try JSONValue(from: decoder)
+
+        if case let .object(jsonObject) = json,
```

Comment:
nit: prefer guard to handle exceptional early exits

Reply:
will keep this in mind but would like to keep as is, this logic will change drastically with the double decoding logic that's performed to support pagination